scenarios/remote_signals/src/main.rs

```diff
@@ -138,15 +136,22 @@ fn agent_behaviour(
                     let admin_client = AdminWebsocket::connect(admin_ws_url, reporter).await?;
                     // No more agents available to signal, get a new list.
                     // This is also the initial condition.
-                    let mut new_peer_list = admin_client
+                    let agent_infos_encoded = admin_client
```

Comment:
Same code comments on this logic as above, extract as common function?

Reply:
Hmmm there was already a bunch of duplication across scenarios, where do you think this common code should live?  in a `scenario_utils` file?

Reply:
There's already a place for this because we have common functions for things like this - `bindings/runner/src/common.rs`

Reply:
moved in ff8610fdd9e8be4c5f3e2d31c1e74a84dda11499

Reply:
Code extraction looks good. A Rust doc comment for the function would be great. The existing functions in that file should have descriptions of how to use them with Rust code examples

Reply:
commend added in 3c7ddf5eab669952dabb380ff007c7823d3ec0c3